### PR TITLE
API can display a list of menu items

### PR DESCRIPTION
**Pull request by emtalen**
PT feature: https://www.pivotaltracker.com/story/show/171466894
```
As a restaurant API
In order to serve a client with products
I would like to be able to provide an endpoint for a products index
```
We added a model and controller for products and set the routes to do a GET request and then see all products. Each product has an ID, category, name, description and price.

### CODE REVIEW

File: app/models/product.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Product < ApplicationRecord
+  validates_presence_of :category_id, :name, :price
```
Comment (tochman) [inline app/models/product.rb]: Please remove `category_id`  for now.

File: config/application.rb [hunk 2]
```diff
@@ -33,6 +29,14 @@ class Application < Rails::Application
           expose: %w(access-token expiry token-type uid client),
           max_age: 0
       end
+      end
```
Comment (tochman) [inline config/application.rb]: Please format your code properly. I can't really read this.

File: config/routes.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
   namespace :api do
+    resources :products, only: [:index]
     namespace :v0 do
       resources :pings, only: [:index], constraints: { format: 'json' }
     end
```
Comment (tochman) [inline config/routes.rb]: If I ware you, I would remove the `v0` name space now.

File: db/migrate/20200303132228_create_products.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class CreateProducts < ActiveRecord::Migration[6.0]
+  def change
+    create_table :products do |t|
+      t.bigint :category_id
```
Comment (tochman) [inline db/migrate/20200303132228_create_products.rb]: Not needed here.

File: db/migrate/20200303132228_create_products.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+      t.string :name
+      t.string :description
+      t.float :price
```
Comment (tochman) [inline db/migrate/20200303132228_create_products.rb:6]: Should that really be a float? Not a blocker

File: spec/factories/products.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :product do
+    category_id { "" }
```
Comment (tochman) [inline spec/factories/products.rb]: This one has to go

File: spec/models/product_spec.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    it  {is_expected.to have_db_column :name}
+    it  {is_expected.to have_db_column :description}
+    it  {is_expected.to have_db_column :price}
+  end
+  describe 'Validations' do
+    it {is_expected.to validate_presence_of :category_id}
+    it {is_expected.to validate_presence_of :name}
+    it {is_expected.to validate_presence_of :price}
+  end
+end
```
Comment (tochman) [inline spec/models/product_spec.rb:19]: Please add a test for the factory

File: spec/requests/api/products_spec.rb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+RSpec.describe Api::ProductsController, type: :request do
+  describe 'GET /product' do
+    before do
+      get '/api/products'
+    end
+    it 'should return a 200 response' do
+      expect(response).to have_http_status 200
+    end
+end
```
Comment (tochman) [inline spec/requests/api/products_spec.rb:19]: Plz mind code formatting.

File: spec/requests/api/products_spec.rb [hunk 9]
```diff
@@ -0,0 +1,10 @@
+RSpec.describe Api::ProductsController, type: :request do
+  describe 'GET /product' do
+    before do
```
Comment (tochman) [inline spec/requests/api/products_spec.rb:7]: Please consider this:

```ruby
let!(:products) { 3.times { FactoryBot.create(:product) } }
```

Then, you can test if the response has data about 3 products.

**Review by tochman — CHANGES_REQUESTED** [review]

File: app/models/product.rb (hunk 1 above)
Comment (emtalen) [reply app/models/product.rb]: Its is removed

**Review by tochman — APPROVED** [review]

**Pull request merged**